app/src/main/java/com/kylecorry/trail_sense/main/MainActivity.kt

```diff
@@ -108,6 +112,25 @@ class MainActivity : AndromedaActivity() {
         }
     }
 
+    fun changeBottomNavLabelVisibility(useCompactMode: Boolean){
+        cache.putBoolean("pref_use_compact_mode", useCompactMode)
+        recreate()
```

Comment:
Is the recreate really needed or can this just call `setBottomNavLabelVisibility` (without recreating the activity)?

Reply:
Yes, you are right, it was a typo on my part because I had originally intended to change the component-specific height, but now changing the layoutparams does not require recreating the activity. Fixed.